tests/integration/Owncloud/OcisPhpSdk/ResourceInviteTest.php

```diff
@@ -0,0 +1,137 @@
+        $shares = $this->resourceToShare->invite([$marie], $this->viewerRole);
+        $this->assertCount(1, $shares);
+        $receivedShares = $marieOcis->getSharedWithMe();
+        $this->assertCount(1, $receivedShares);
+        $this->assertSame($this->resourceToShare->getName(), $receivedShares[0]->getName());
+    }
+
+    public function testInviteMultipleUsersAtOnce(): void
+    {
+        $marieOcis = $this->initUser('marie', 'radioactivity');
+        $marie = $this->ocis->getUsers('marie')[0];
+        $shares = $this->resourceToShare->invite([$this->einstein,$marie], $this->viewerRole);
+        $this->assertCount(2, $shares);
+        $receivedShares = $marieOcis->getSharedWithMe();
+        $this->assertCount(1, $receivedShares);
+        $this->assertSame($this->resourceToShare->getName(), $receivedShares[0]->getName());
+
+        $receivedShares = $this->einsteinOcis->getSharedWithMe();
+        $this->assertCount(1, $receivedShares);
+        $this->assertSame($this->resourceToShare->getName(), $receivedShares[0]->getName());
+    }
+
+    public function testInviteGroup(): void
+    {
+        $philosophyHatersGroup =  $this->ocis->createGroup(
+            'philosophy-haters',
+            'philosophy haters group'
+        );
+        $this->createdGroups = [$philosophyHatersGroup];
+        $philosophyHatersGroup->addUser($this->einstein);
+        $shares = $this->resourceToShare->invite([$philosophyHatersGroup], $this->viewerRole);
+        $this->assertCount(1, $shares);
+        $receivedShares = $this->einsteinOcis->getSharedWithMe();
+        $this->assertCount(1, $receivedShares);
+        $this->assertSame($this->resourceToShare->getName(), $receivedShares[0]->getName());
+    }
+
+    public function testInviteGroupAndUserOfTheGroup(): void
+    {
+        $philosophyHatersGroup =  $this->ocis->createGroup(
+            'philosophy-haters',
+            'philosophy haters group'
+        );
+        $this->createdGroups = [$philosophyHatersGroup];
+        $philosophyHatersGroup->addUser($this->einstein);
+        $shares = $this->resourceToShare->invite([$philosophyHatersGroup, $this->einstein], $this->viewerRole);
+        $this->assertCount(2, $shares);
+        $receivedShares = $this->einsteinOcis->getSharedWithMe();
+        $this->assertCount(2, $receivedShares);
+        $this->assertSame($this->resourceToShare->getName(), $receivedShares[0]->getName());
+        $this->assertSame($this->resourceToShare->getName(), $receivedShares[1]->getName());
+    }
+
+    public function testInviteSameUserAgain(): void
+    {
+        $this->markTestSkipped('https://github.com/owncloud/ocis/issues/7842');
+        // @phpstan-ignore-next-line because the test is skipped
+        $this->expectException(ForbiddenException::class);
+        $this->resourceToShare->invite([$this->einstein], $this->viewerRole);
+        $this->resourceToShare->invite([$this->einstein], $this->viewerRole);
```

Comment:
how about using anotherRole for second invite
```suggestion
        $this->resourceToShare->invite([$this->einstein], $this->viewerRole);
        $this->resourceToShare->invite([$this->einstein], $this->anotherRole);
```